doc/SDK_NOTIFICATIONS.md

```diff
@@ -0,0 +1,77 @@
+# Breez SDK Notifications
+Applications have their own way of delivering notifications to end users such as: mobile push, web push, emails, text messages etc...
+This is usually implemented using some backend service that has a way to communicate with the end user. The SDK goal is to provide the trigger including the type and notification data to that service. A simple and standard option to implement that trigger is using webhooks.
+
+## The use cases:
+
+### Offline payment
+User is offline and would like to be notified to accept a received payment.
+- The client associate its node id with the webhook by calling a dedicated API endpoint of the LSP.
+- The sender pays, htlcs land at the LSP which detects the last hop is not connected as a peer and send a POST request to the associated webhook with the payload that includes the notification type and data.
+- The web hook notifies the delivery service that builds the specific notification format and payload and send it to the device. In that case the push message can be silent, waking up the app to run the signer and process the payment.
+
+### Receive on-chain
+User is offline and would like to be notified when the transaction used to send funds to an on-chain swap address is confirmed.
+- The client initiate a swap and send funds to the on-chain address.
+- The client registers to a push notification for the address using web hook.
+- Once the transaction is confirmed the notification server send a POST request to the associated webhook with the payload that includes the notification type and data.
+- The web hook notifies the delivery services that construct the specific notification format and payload and send it to the device. In that case the push message can be silent, waking up the app to run the signer and complete the swap payment.
+
+## Suggested implementation
+There are several parts for achieving the above:
+
+### SDK
+- `register_notifications(url, platform, token)` API method for the user to register all kind of notifications.
```

Comment:
Any webhook subscription should have an expiry time, to avoid having infinite dangling subscriptions. (maybe we could eeven notify when the expiry time is coming up)

Reply:
I was thinking the expiry time is subject to the service policy. For example the service may decide that if the client didn't refresh the registration for 3 days the registration would expire. 



> maybe we could even notify when the expiry time is coming up

Nice. Notification (in mobile could be even silent) to refresh the registration seems a great option to achieve this goal UX wise.